File: 백준/Silver/20006. 랭킹전 대기열/랭킹전 대기열.c
#include <stdio.h>
#include <string.h>

int main() {
    int p, m;
    int level[300];
    int room[300][300];
    char name[300][17];
    int room_count = 0
    int room_size[300] = {0};

    // 입력 받기
    scanf("%d %d", &p, &m);
    for (int i = 0; i < p; i++) {
        scanf("%d %s", &level[i], name[i], 17);
    }

    // 플레이어 입장 처리
    for (int i = 0; i < p; i++) {
        int placed = 0;
        for (int j = 0; j < room_count; j++) {
            if (room_size[j] < m && level[room[j][0]] - 10 <= level[i] && level[i] <= level[room[j][0]] + 10) {
                room[j][room_size[j]++] = i;
                placed = 1;
                break;
            }
        }
        if (!placed && room_count < 300) {
            room[room_count][0] = i;
            room_size[room_count++] = 1;
        }
    }

    // 출력
    for (int i = 0; i < room_count; i++) {
        if (room_size[i] == m) {
            printf("Started!\n");
        } else {
            printf("Waiting!\n");
        }

        // 닉네임 정렬 (삽입 정렬 사용)
        for (int j = 1; j < room_size[i]; j++) {
            for (int k = j; k > 0 && strcmp(name[room[i][k - 1]], name[room[i][k]]) > 0; k--) {
                int temp = room[i][k];
                room[i][k] = room[i][k - 1];
                room[i][k - 1] = temp;
            }
        }

        // 플레이어 출력
        for (int j = 0; j < room_size[i]; j++) {
            int player_index = room[i][j];
            printf("%d %s\n", level[player_index], name[player_index]);
        }
    }
    return 0;
}

File: 백준/Silver/20006. 랭킹전 대기열/랭킹전 대기열.py
p, m = map(int, input().split())

rooms = []  

for _ in range(p):
    level, name = input().split()
    if len(rooms) == 0:     
        rooms.append([[int(level), name]])    
        
    else:    
        for room in rooms:      
            first_lv = room[0][0]     
            if len(room) < m and first_level - 10 <= int(level) <= first_level + 10:
                room.append([level, name])  
                break     
                
        else:     
            rooms.append([[int(level), name]])     
    
for room in rooms:
    if len(room) == m:
        print('Started!')
    else:
        print('Waiting!')
    room.sort(key=lambda x:x[1]) 
    
    for player in room:
        print(player[0], player[1])
